iOS-23 | Добавлена логика экрана секций

mightyK1ngRichard opened the pull request:
## Что было сделано:
* Создан firbase сервис для категорий
* Поправлен UI
* Рефакторинг моделей
* Добавлен роутинг

## TODO:
- [X] Кэширование

<img src="https://github.com/iOS-vk-education/2023_2_CakesHub/assets/90796324/2ecec150-36f9-4f40-bca6-37746b386328" width="250px">

mightyK1ngRichard committed:
iOS-4: create templates for swiftui mvvm

mightyK1ngRichard committed:
Update Xcode Templates/VK-CakesHub/SwiftUI Component.xctemplate/CHM___FILEBASENAME___/Configuration/CHM___FILEBASENAME___+Defaults.swift

Co-authored-by: MatveyGarbuzov <75742016+MatveyGarbuzov@users.noreply.github.com>

mightyK1ngRichard committed:
Update Xcode Templates/VK-CakesHub/SwiftUI Component.xctemplate/CHM___FILEBASENAME___/Configuration/CHM___FILEBASENAME___+Defaults.swift

Co-authored-by: MatveyGarbuzov <75742016+MatveyGarbuzov@users.noreply.github.com>

mightyK1ngRichard committed:
iOS-4: minor

Вынес в константу

mightyK1ngRichard committed:
iOS-4: Обновление шаблона

mightyK1ngRichard committed:
iOS-4: update view template

blackplayerten committed:
Merge pull request #1 from iOS-vk-education/feature/permyakov/iOS-1_create_application

iOS-1 | Создание проекта

mightyK1ngRichard committed:
Merge branch 'main' into tech_debt/permyakov/iOS-4_create_mvvm_templates

# Conflicts:
#	.gitignore

blackplayerten committed:
Merge pull request #2 from iOS-vk-education/tech_debt/permyakov/iOS-4_create_mvvm_templates

Добавление шаблонов для создания экрана и компонента ДС

mightyK1ngRichard committed:
iOS-13: Добавил readme

blackplayerten committed:
Merge pull request #23 from iOS-vk-education/tech_debt/permyakov/iOS-13_add_readme

iOS-13 | Добавление README файла

mightyK1ngRichard committed:
iOS-23: Создан сервис категорий

mightyK1ngRichard committed:
iOS-23: Добавил фетч по тегам

mightyK1ngRichard committed:
iOS-23: Добавлены запросы в сеть и фильтры

1. Добавил запросы в сеть у экрана категорий
2. Переработал UI
3. Добавил фильтрацию
4. Обновил View Model

mightyK1ngRichard committed:
iOS-23: Закончил экран секций

mightyK1ngRichard committed:
iOS-23: Добавил кэширование категорий

1. Добавил кэширование данных категорий
2. SDModelable добавил mapper в FBModelable

mightyK1ngRichard committed:
iOS-24: Добавлены копирайты

mightyK1ngRichard committed:
iOS-24: Сверстал компонент ячейки чата

mightyK1ngRichard committed:
iOS-24: Начат экран чатов пользователя

mightyK1ngRichard committed:
iOS-24: Добавлен логика экрана всех чатов

mightyK1ngRichard committed:
iOS-24: minor

mightyK1ngRichard committed:
iOS-24: minor

mightyK1ngRichard committed:
iOS-24: Добавил переход на экран чатов

mightyK1ngRichard committed:
Merge branch 'main' into feature/permyakov/iOS-24_chats_history

mightyK1ngRichard committed:
iOS-24: Правки моделей

mightyK1ngRichard committed:
iOS-24: Правки моделей

mightyK1ngRichard committed:
iOS-25: Миграция на NotificationCenter

mightyK1ngRichard committed:
iOS-25: Полный фикс веб сокет ивентов

Обнаружена и пофикшена проблема получения данных по веб сокет соединению. Оказалось, что на двух разных экранах был один и тот же обработчик события получения ифнормации. Но один получал сообщения, а второй уведомления, и если мы переходили на оба экрана, главный обработчик из синглтон метода парсил, чередуясь между сообещние и уведомлением, хотя приходило сообщение.
Теперь обработчки событий один и он вызывается при запуске приложения, а объекты отправляет через Notification Center

mightyK1ngRichard committed:
iOS-26: Добавил кэширование сообщений

mightyK1ngRichard committed:
iOS-26: Добавил комментарии к коду

mightyK1ngRichard committed:
iOS-26: Мелкие фиксы

mightyK1ngRichard committed:
iOS-26: Добавлен запрос истории чата из профиля

itissep commented on CakesHub/CakesHub/Application/MainView/ViewModel/RootViewModel.swift:
```diff
@@ -61,7 +61,7 @@ extension RootViewModel: RootViewModelProtocol {
 
         // Достаём закэшированные данные
         let sdProducts = fetchProductsFromMemory()
-        let fbProducts = sdProducts.map { $0.mapperInFBProductModel }
```
🔥

itissep commented on CakesHub/CakesHub/Application/Navigation.swift:
```diff
@@ -30,23 +30,23 @@ extension Navigation {
 
     var title: String {
         switch self {
         case .house:
```
может и house переименовать в home?

itissep commented on CakesHub/CakesHub/Services/Firebase/Entities/FBCateoryModel/FBCategoryTag.swift:
```diff
@@ -0,0 +1,17 @@
+
+import Foundation
+
+extension FBCateoryModel {
```
Cate g ory

mightyK1ngRichard committed:
iOS-27: Добавлено кэширование фоток. Обновление моделей

mightyK1ngRichard committed:
iOS-27: Удалил тестовый экран

itissep commented on CakesHub/CakesHub/Streams/ProductStream/CategoriesScreen/ViewModel/CategoriesViewModel.swift:
```diff
@@ -6,42 +6,173 @@
+                let fbSections = try await fetch()
+                withAnimation {
+                    sections = [
+                        .men(fbSections[0].items.sorted(by: { $0.title < $1.title }).mapper),
+                        .women(fbSections[1].items.sorted(by: { $0.title < $1.title }).mapper),
+                        .kids(fbSections[2].items.sorted(by: { $0.title < $1.title }).mapper)
```
комментарий скорее к дизайну. мужские тортики точно должны быть первой секцией? как будто по популярности порядок будет женские, детские, мужские?

mightyK1ngRichard replied:
сделал женские на первом)

mightyK1ngRichard committed:
iOS-28: Создал экран создания отзыва

mightyK1ngRichard committed:
iOS-28: Добавил логику для экрана отзыва

mightyK1ngRichard committed:
iOS-28: Доработка экрана отзыва и моделей

1. Добавил обновление информации о товаре при оставке отзыва
2. Поправил расчёты логики рейтинга товара

mightyK1ngRichard committed:
iOS-28: Убрал анимацию при обновлении категорий

mightyK1ngRichard committed:
iOS-29: Добавил локальные пуши с задержкой

mightyK1ngRichard committed:
iOS-30: Обновил шаблон

mightyK1ngRichard committed:
iOS-30: Добавлена карта

mightyK1ngRichard committed:
iOS-30: minor

mightyK1ngRichard committed:
iOS-30: Minor

mightyK1ngRichard committed:
iOS-30: fix bug

mightyK1ngRichard committed:
iOS-31: Добавил архитектуру для экрана насроек и логику выхода

mightyK1ngRichard committed:
iOS-31: Доработка AuthView под новый подход

mightyK1ngRichard committed:
iOS-31: minor

mightyK1ngRichard committed:
Merge pull request #43 from iOS-vk-education/feature/permyakov/iOS-31_logout

iOS-31 | Добавлена архитектура для экрана настроек. Внедрена логика выхода их аккаунта

mightyK1ngRichard committed:
Merge pull request #42 from iOS-vk-education/feature/permyakov/iOS-30_map_view

iOS-30 | Добавлена карта

mightyK1ngRichard committed:
Merge pull request #41 from iOS-vk-education/feature/permyakov/iOS-29_add_local_push_notifications

iOS-29 | Добавлены локальные пуши

mightyK1ngRichard committed:
Merge pull request #40 from iOS-vk-education/feature/permyakov/iOS-28_add_feedback_screen

iOS-28 | Добавлен экран фидбэка о товаре

mightyK1ngRichard committed:
Merge pull request #39 from iOS-vk-education/feature/permyakov/iOS-27_image_cache

iOS-27 | Добавил кэширование фотографий в FileManager

mightyK1ngRichard committed:
Merge pull request #38 from iOS-vk-education/feature/permyakov/iOS-26_chat_history

iOS-26 | Добавлено кэширование сообщений

mightyK1ngRichard committed:
Merge pull request #37 from iOS-vk-education/bug/permyakov/iOS-25_fix_web_socket_link

iOS-25 | Поправлен Web Socket обработчик

mightyK1ngRichard committed:
Merge pull request #36 from iOS-vk-education/feature/permyakov/iOS-24_chats_history

iOS-24 | Создан экран всех чатов
